pg_search/src/postgres/customscan/pdbscan/solve_expr.rs

```diff
@@ -4,22 +4,35 @@ use pgrx::{pg_sys, FromDatum, PgMemoryContexts};
 
 impl SearchQueryInput {
     pub fn has_postgres_expressions(&mut self) -> bool {
-        for sqi in self {
-            if matches!(sqi, SearchQueryInput::PostgresExpression { .. }) {
-                return true;
+        let mut found = false;
+        self.visit(&mut |sqi| {
+            match sqi {
+                SearchQueryInput::PostgresExpression { .. } => {
+                    found = true;
+                }
+                SearchQueryInput::HeapFilter { field_filters, .. } => {
+                    // Check if any heap field filters contain subqueries
+                    for filter in field_filters.iter() {
+                        if filter.contains_subqueries() {
+                            found = true;
+                            break;
+                        }
+                    }
+                }
```

Comment:
We're doing this extra check in `has_postgres_expressions`, but then we're not doing anything special with `HeapFilter` nodes in `init_postgres_expressions` and `solve_postgres_expressions`... why is that? Might be worth a comment.

Reply:
Good observation. After more thinking, made sense to have its own method. Done.